File: app/src/main/java/pl/szczodrzynski/edziennik/datamodels/FeedbackMessage.java
package pl.szczodrzynski.edziennik.datamodels;

import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "feedbackMessages")
public class FeedbackMessage {

    @PrimaryKey(autoGenerate = true)
    public int messageId;

    public boolean received = false;
    public String fromUser = null;
    public String fromUserName = null;
    public long sentTime = System.currentTimeMillis();
    public String text;

    public FeedbackMessage(boolean received, String text) {
        this.received = received;
        this.sentTime = System.currentTimeMillis();
        this.text = text;
    }

    @Ignore
    public FeedbackMessage() {
    }
}

File: app/src/main/java/pl/szczodrzynski/edziennik/data/db/entity/FeedbackMessage.kt
/*
 * Copyright (c) Kuba Szczodrzyński 2020-1-21.
 */

package pl.szczodrzynski.edziennik.data.db.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "feedbackMessages")
open class FeedbackMessage(
        @PrimaryKey(autoGenerate = true)
        val messageId: Int = 0,

        val received: Boolean,
        val text: String,

        // used always - contains the sender name
        val senderName: String,

        // used in DEV apps - contains device ID and model
        var deviceId: String? = null,
        var deviceName: String? = null,

        val devId: Int? = null,
        val devImage: String? = null,

        val sentTime: Long = System.currentTimeMillis()
) {
        class WithCount(messageId: Int, received: Boolean, text: String, senderName: String, deviceId: String?, deviceName: String?, devId: Int?, devImage: String?, sentTime: Long) : FeedbackMessage(messageId, received, text, senderName, deviceId, deviceName, devId, devImage, sentTime) {
                var count = 0
        }
}
